tests/test_main.py

```diff
@@ -289,67 +292,57 @@ def test_initialize_github_error(
         mock_report_failure.assert_called_with(self.mock_config)
 
     @mock.patch(PATH + "handle_msg")
-    @mock.patch(PATH + "fedmsg")
-    def test_listen_no_handlers(self, mock_fedmsg, mock_handle_msg):
+    @mock.patch(PATH + "load_config")
+    def test_listen_no_handlers(self, mock_load_config, mock_handle_msg):
         """
         Test 'listen' function where suffix is not in handlers
         """
         # Set up return values
-        mock_fedmsg.tail_messages.return_value = [
-            ("dummy", "dummy", "mock_topic", self.old_style_mock_message)
-        ]
+        mock_load_config.return_value = self.mock_config
 
         # Call the function
-        m.listen(self.mock_config)
+        self.old_style_mock_message.topic = "mock_topic"
+        m.callback(self.old_style_mock_message)
 
         # Assert everything was called correctly
         mock_handle_msg.assert_not_called()
 
     @mock.patch(PATH + "handle_msg")
     @mock.patch(PATH + "issue_handlers")
-    @mock.patch(PATH + "fedmsg")
-    def test_listen_no_issue(self, mock_fedmsg, mock_handlers_issue, mock_handle_msg):
+    @mock.patch(PATH + "load_config")
+    def test_listen_no_issue(
+        self, mock_load_config, mock_handlers_issue, mock_handle_msg
+    ):
         """
         Test 'listen' function where the handler returns none
         """
         # Set up return values
         mock_handlers_issue["github.issue.comment"].return_value = None
-        mock_fedmsg.tail_messages.return_value = [
-            ("dummy", "dummy", "d.d.d.github.issue.drop", self.old_style_mock_message)
-        ]
+        mock_load_config.return_value = self.mock_config
 
         # Call the function
-        m.listen(self.mock_config)
+        self.old_style_mock_message.topic = "d.d.d.github.issue.drop"
```

Comment:
Given that the suffix will presumably be `github.issue.drop`, why are we patching the entry for `github.issue.comment`?

In any case, I think that the reason why this test is passing is because `github.issue.drop` doesn't match any handler...so I don't think this test is functioning as intended (as coded, I think it's redundant with the previous test).

Reply:
Agreed - let's delete this test.